src/Encamina.Enmarcha.AspNet.OpenApi.Swashbuckle/SchemaFilters/AtLeastOneRequiredSchemaFilter.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Reflection;
+
+using Encamina.Enmarcha.Core.DataAnnotations;
+
+using Microsoft.OpenApi.Models;
+
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace Encamina.Enmarcha.AspNet.OpenApi.Swashbuckle.SchemaFilters;
+
+/// <summary>
+/// This filter enforces that at least one of a specified set of properties must be required in an OpenAPI schema.
+/// It processes types decorated with the <see cref="AtLeastOneRequiredAttribute"/>.
+/// </summary>
+public sealed class AtLeastOneRequiredSchemaFilter : ISchemaFilter
+{
+    /// <summary>
+    /// Applies the filter to modify the OpenAPI schema based on the <see cref="AtLeastOneRequiredAttribute"/>.
+    /// It removes individual property requirements and replaces them with an 'anyOf' requirement
+    /// for the specified properties, ensuring that at least one of them is present in the request.
+    /// </summary>
+    /// <param name="schema">The <see cref="OpenApiSchema"/> to modify.</param>
+    /// <param name="context">
+    /// The <see cref="SchemaFilterContext"/> containing metadata about the context, including the type being processed.
+    /// </param>
+    public void Apply(OpenApiSchema schema, SchemaFilterContext context)
+    {
+        var attribute = context.Type.GetCustomAttribute<AtLeastOneRequiredAttribute>();
+        if (attribute != null)
+        {
+            // Ensure 'Required' is not null
+            schema.Required ??= new HashSet<string>();
+
+            // Remove individually required properties specified in the attribute
+            foreach (var propName in attribute.PropertyNames)
+            {
+                schema.Required.Remove(propName);
```

Comment:
@MarioRamosEs don't we have a better method to remove all? avoiding the foreach? make sense=

Reply:
From my understanding, does it make sense for a property to have both `RequiredAttribute` and `AtLeastOneRequiredAttribute`? I feel like this is contradictory. If I’m not mistaken, we shouldn’t remove the `Required` from those properties in the schema since they shouldn't both exist. 

*Disclaimer: I might be missing some context here :)*

Reply:
So, if the `foreach` is noisy we could do this:

```csharp
schema.Required.ExceptWith(new HashSet<string>(attribute.PropertyNames));
```

But @albertodiazencamina an iteration over the elements is going to happen nonetheless, just under the hood.

Regarding @LuisM000 comment, yes... it kinda of dosen't make sense. If something is `Required` is because it must be always provided, either by parameter or by default. We should use the `AtLeastOneRequiredAttribute` when we could have various required properties but the logic is satisfied with at least just one. 

Now I don't know where the comment from @LuisM000  comes from 😜😜😜

Reply:
To clarify, I don’t see the need to remove the Required properties because they shouldn’t coexist with AtLeastOneRequiredAttribute. Having both creates a conflict, as Required implies that a property must always be present, which contradicts the logic of requiring at least one from a set.

If I’m just causing noise here, feel free to ignore my comment, everyone! :)

Reply:
@MarioRamosEs please could you share clarifications about this?

Perhaps a unit test or something to illustrate this behavior? 😀